### PR TITLE
Publish catalogue date

### PR DESCRIPTION
**Pull request by MightyAx**
The commit text got changed but:

- [x] Remove publishedToCatalogue
- [x] Add Boolean isInCatalogue
- [x] Add Instant cataloguedDate
- [x] Remove any set of cataloguedDate sent by clients
- [x] set cataloguedDate to now when isInCatalogue is set to True
- [x] Push logic into ProjectService.

### CODE REVIEW

**Review by jacobwindsor — APPROVED** [review]

**Review by jacobwindsor — APPROVED** [review]

**Review by yusra-haider — APPROVED** [review]

File: src/main/java/org/humancellatlas/ingest/project/ProjectService.java [hunk 1]
```diff
@@ -61,11 +64,31 @@ protected Logger getLog() {
...
         Project persistentProject = projectRepository.save(project);
         projectEventHandler.registeredProject(persistentProject);
         return persistentProject;
     }
 
+    public Project update(final Project project, ObjectNode patch, Boolean sendNotification) {
+        if (patch.has("isInCatalogue")
+            && patch.get("isInCatalogue").asBoolean()
+            && project.getCataloguedDate() == null){
+            project.setCataloguedDate(Instant.now());
```
Comment (aaclan-ebi) [inline src/main/java/org/humancellatlas/ingest/project/ProjectService.java:80]: If the checkbox was checked after being unchecked, the catalogue date will always contain the date when it was first checked. Is that what we want? 

If not, maybe we could only update the catalogue date when isInCatalogue is false and being set to true?

Comment (clairerye) [reply src/main/java/org/humancellatlas/ingest/project/ProjectService.java:80]: I think this is probably an edge case. Its could that we can remove projects from the catalogue if the box is unchecked, but in the scenario that it is "in the catalogue" then "out of the catalogue" then "in the catalogue" again, I think its ok to use the original "in" date. I hope that makes sense? So I think its fine as is...

**Review by aaclan-ebi — APPROVED** [review]
Looks good!

File: src/main/java/org/humancellatlas/ingest/project/ProjectService.java (hunk 1 above)
Comment (aaclan-ebi) [reply src/main/java/org/humancellatlas/ingest/project/ProjectService.java:80]: I see, thanks!

Comment (clairerye) [reply src/main/java/org/humancellatlas/ingest/project/ProjectService.java:80]: great thinking on this Alegria! Its great how many scenarios you are thinking through here

**Pull request merged**